docs/tokens/lockup.md

```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
+
+The lockups are implemented as a separate smart contract from your main account.
+Thus, if you have received tokens prior to [Phase II](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/), you will get two things:
+
+1. A regular account (also called "Owner Account" in context of lockups), let's say `user.near` or `3e52c197feb13fa457dddd102f6af299a5b63465e324784b22aaa7544a7d55fb`;
+2. A lockup contract, with name `4336aba00d32a1b91d313c81e8544ea1fdc67284.lockup.near`.
+
+The owner account is created first, either by following the NEAR Drop process or by creating new key pair using Trust, Ledger or another wallet.
+
+The lockup contract is then deployed with a predictable name.
+It is defined as `hash(owner_account_id)[:20]` encoded in `hex` and deployed as subaccount under `lockup.near`.
+It means that all lockup contracts end with `.lockup.near`.
+
+If you have added or earned additional tokens above the initial balance, they are considered unlocked and can be freely withdrawn.
+For example, any rewards that are earned using this lockup contract (eg from delegation) or any other funds sent to this lockup contract can be withdrawn by the owner at any time.
+
+The actual lockup release process happens on per block basis. 
+E.g. if the release length is 1 calendar year, it will actually be `31,536,000` seconds, and with ~1 second blocks, `~1/31,536,000` will be released per block.
+When the lockup has been fully released, the Owner Account can add the full-access key and withdraw all the funds from it.
+
+The contract consists of lockup and vesting processes that go simultaneously.
+Both of these processes lock the tokens, but the mechanics slightly differ.
+
+### Lockup
+
+Lockup mechanics have 2 configurable parameters: 
+1. `lockup_timestamp` - The moment when tokens start linearly unlocking;
+2. `release_duration` - The length of the unlocking schedule during which tokens are linearly unlocked.
+  By the end of this duration, all tokens are unlocked.
+
+The lockup process could not be terminated.
+Lockup does not have a cliff.
+
+![lockup_1](/docs/assets/lockup/lockup_1.png)
+
+[deprecated] Apart from the lockup timestamp, there is a lockup duration.
+`lockup_duration` is the interval between [the Phase II launch](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/) (October 13th) and the moment when tokens start to unlock.
+
+![lockup_2](/docs/assets/lockup/lockup_2.png)
+
+### Vesting
 
-## The Lockup Contract
+Vesting also locks the tokens, and it allows to configure 2 more options:
+1. Ability to terminate tokens vesting and refund non-vested tokens back.
+2. Cliff vesting period.
 
-`Lockup` is a special smart contract that ensures that the full amount or even a partial amount is not transferable until it is supposed to be.
+Vesting process contains of 3 timestamps: `start_date`, `cliff_date`, `end_date`.
```

Comment:
```suggestion
Vesting process is defined by 3 timestamps: `start_date`, `cliff_date`, `end_date`.
```

Reply:
Changed it to `includes` because `is defined by` sounds for me like if I have these 3 timestamps, I should be able to say which vesting process it is.